### ISSUE TITLE
Tabular view with NHS Icons

### ISSUE THREAD

**Issue by moranm1** [AUS-DOH-Safety-and-Quality/PowerBI-SPC#293]
Please would it be possible to include the ability to display data in a tabular view with the variation and assurance icons as per the example below? This is the recommended approach across Integrated Performance Reports for a lot of NHS Trusts.

![image](https://github.com/AUS-DOH-Safety-and-Quality/PowerBI-SPC/assets/131153378/a3030977-957f-4e49-bead-5c890abdccff)

**Comment by carolineb936** [comment]
Yes Please would love this

**Comment by LeslieHewitt** [comment]
This is in the [https://future.nhs.uk/MDC/view?objectId=118528357](power BI Template) on the FutureNHS website.

**Comment by andrjohns** [comment]
A summary table with icons is now implemented ([see guide here](https://aus-doh-safety-and-quality.github.io/PowerBI-SPC/SummaryTable.html) for more details)

**Comment by moranm1** [comment]
Hi thank you for implementing this, which version of the tool is it available in? The version on the Microsoft store is 1.4.1.4, I've downloaded it and I cant see the Grouping (summary table) field.
Thanks

**Comment by andrjohns** [comment]
It's currently only available in the development version of the visual ([installation instructions here](https://aus-doh-safety-and-quality.github.io/PowerBI-SPC/Installation.html))

We're currently completing a round of testing and gathering feedback on the implementation, before submitting to Microsoft in a month or two (depending on feedback)

**Comment by carolineb936** [comment]
OK am I understanding this correctly we cannot test it yet?

**Comment by carolineb936** [comment]
I have imported and tried this with my date and get an error when trying to add the NHS icons

**Comment by carolineb936** [comment]
![image](https://github.com/user-attachments/assets/f812f68e-97ce-458f-a5d0-3f52c192729c)

**Comment by moranm1** [comment]
I got the same error you have to make sure you have turned on the Outlier Detection settings

**Comment by andrjohns** [comment]
> I have imported and tried this with my date and get an error when trying to add the NHS icons

Yes, as the error message indicates, the Variation icon is meaningless if you haven't enabled the detection of any outlier patterns. 

The Variation icon is used to indicate the presence of special-cause or common-cause variation, where special-cause variation is determined by the presence of an outlier pattern (astronomical point, shift, trend, etc.). If you haven't enabled the detection of any outlier patterns (i.e., the detection of special-cause variation), then the variation icon will always show common cause

**Comment by carolineb936** [comment]
Is there a limit to the number of rows showing - I can only see 10 - not sure if I can show more
![image](https://github.com/user-attachments/assets/2ad27beb-0071-452c-b7cd-b6d85bb7c5eb)

**Comment by andrjohns** [comment]
> Is there a limit to the number of rows showing - I can only see 10 - not sure if I can show more ![image](https://private-user-images.githubusercontent.com/170107144/356157711-2ad27beb-0071-452c-b7cd-b6d85bb7c5eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxMDg1MjgsIm5iZiI6MTcyMzEwODIyOCwicGF0aCI6Ii8xNzAxMDcxNDQvMzU2MTU3NzExLTJhZDI3YmViLTAwNzEtNDUyYy1iN2NkLWI2ZDg1YmI3YzVlYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwOFQwOTEwMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YzI3OWY0YzQ1OTAwM2ZjOGY0MDBiMjNjYzdkYmE2MDU5NjZjZTNlNjAzMDY1NTQ5YzllZDU3ZWI2NWMwZTJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6TbcGz6dZOg-Mc2gvH3BpIE8rWm_uMDAZH4rky8QJgg)

How many rows of data are you passing to the visual? PowerBI has a hard limit of [30 000 rows for custom visuals](https://learn.microsoft.com/en-us/power-bi/developer/visuals/dataview-mappings#data-reduction-algorithm-types)

There's an API to add support for [fetching additional rows](https://learn.microsoft.com/en-us/power-bi/developer/visuals/fetch-more-data), but it requires some additional complexity in data management and fetching/appending updates, so I don't think I'll be able to add it for the next release

**Comment by carolineb936** [comment]
It is only a few more rows I was after (eg 20) But I will filter them into groups of 10 and show 2 tables instead, until an update comes through

**Comment by andrjohns** [comment]
> It is only a few more rows I was after (eg 20) But I will filter them into groups of 10 and show 2 tables instead, until an update comes through

Just to clarify - it's not about the number of rows displayed in the table (10 vs 20), but the amount of data being passed to the visual (e.g., the length of the `numerator` column) - so you might still receive the same warning when filtering to groups of 10 if the resulting data exceeds 30 000 rows

**Comment by carolineb936** [comment]
Sorry If I am being a bit dense but if I filter on top 10 and bottom 10 it works OK and the underlying data rows are the same
